spring-integration-core/src/test/java/org/springframework/integration/json/SimpleJsonSerializerTests.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.json;
+
+import static org.hamcrest.CoreMatchers.containsString;
+import static org.hamcrest.CoreMatchers.endsWith;
+import static org.hamcrest.CoreMatchers.startsWith;
+import static org.hamcrest.Matchers.arrayWithSize;
+import static org.junit.Assert.assertThat;
+
+import org.junit.Test;
+
+/**
+ * @author Gary Russell
+ * @since 5.0
+ *
+ */
+public class SimpleJsonSerializerTests {
+
+	@Test
+	public void test() {
+		Foo foo = new Foo();
+		String json = SimpleJsonSerializer.toJson(foo, "foo", "bar", "dub", "bool");
+		assertThat(json, startsWith("{"));
+		assertThat(json, containsString("\"bool\":true"));
+		assertThat(json, containsString("\"bar\":42"));
+		assertThat(json, containsString("\"foo\":\"bar\""));
+		assertThat(json, containsString("\"dub\":1.6"));
+		assertThat(json, endsWith("}"));
+		String[] split = json.split(",");
+		assertThat(split, arrayWithSize(4));
```

Comment:
??? Why don't test that it is fully compatible with JSON expected by the Jackson for example?
I mean serialize our way, but deserialize to `Foo` with Jackson.